File: examen/app/src/main/java/com/example/examen1/Modelo.kt
package com.example.examen1

class Modelo(
    var nombre: String,
    var anio_lanzamiento: Int,
    var precio: Float,
    var disponible: Boolean,//estado
    var marca: String,
    var latitud: String,
    var longitud: String,
    var url: String,
    var urlImg: String
){
    override fun toString(): String {
        return "Modelo(Nombre='$nombre', Año Lanzamiento=$anio_lanzamiento, Precio=$precio, Disponible=$disponible, Marca='$marca')"
    }
}

File: 03-backend/mascotas/api/models/Modelo.js
/**
 * Modelo.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {
    nombre: {
      type:"string",
      unique: true,
      required: true
    },
    anio_lanzamiento:{
      type:"number",
      required: true
    },
    precio: {
      type:"number",
      required: true
    },
    disponible: {
      type:"boolean",
      required: true
    },
    latitud: {
      type:"string"
    },
    longitud: {
      type:"string"
    },
    url: {
      type:"string"
    },
    url_img: {
      type:"string"
    },
    marca: { // Many to One (nombre FK) - mismo nombre q la relacion
      model: 'marca',
      required: true
      // required: true // (Es opcional 1 muchos 0 muchos)
    } 
  },

};

